src/bridges/convex/ConvexStakingBridge.sol

```diff
@@ -0,0 +1,258 @@
+    error UnknownVirtualAsset();
+    error IncorrectInteractionValue(uint256 stakedValue, uint256 valueToWithdraw);
+
+    constructor(address _rollupProcessor) BridgeBase(_rollupProcessor) {
+        _loadPools();
+
+        uint256[] memory criterias = new uint256[](2 * lastPoolLength);
+        uint32[] memory gasUsage = new uint32[](2 * lastPoolLength);
+        uint32[] memory minGasPerMinute = new uint32[](2 * lastPoolLength);
+
+        for (uint256 i = 0; i < lastPoolLength; i++) {
+            (address curveLpToken, , , , , ) = BOOSTER.poolInfo(i);
+            criterias[i] = uint256(keccak256(abi.encodePacked(curveLpToken, address(0))));
+            gasUsage[i] = 1000000;
+            minGasPerMinute[i] = 690;
+
+            criterias[lastPoolLength + i] = uint256(keccak256(abi.encodePacked(address(0), curveLpToken)));
+            gasUsage[lastPoolLength + i] = 375000;
+            minGasPerMinute[lastPoolLength + i] = 260;
+        }
+
+        SUBSIDY.setGasUsageAndMinGasPerMinute(criterias, gasUsage, minGasPerMinute);
+    }
+
+    /**
+    @notice function so the bridge can receive ether. Used for subsidy.
+    */
+    receive() external payable {}
+
+    /**
+    @notice Stake and unstake Curve LP tokens through Convex Finance Booster anytime.
+    @notice Convert rate between Curve LP token and corresponding Convex LP token is 1:1
+    @notice Stake == Deposit, Unstake == Withdraw
+    @param _inputAssetA Curve LP token (staking), virtual asset (unstaking)
+    @param _outputAssetA Virtual asset (staking), Curve LP token (unstaking)
+    @param _totalInputValue Total number of Curve LP tokens to deposit / withdraw
+    @param _auxData Data to claim staking rewards
+    @param outputValueA Number of Curve LP tokens staked / unstaked
+    @param _rollupBeneficiary Address of the contract that receives subsidy
+    */
+    function convert(
+        AztecTypes.AztecAsset calldata _inputAssetA,
+        AztecTypes.AztecAsset calldata _inputAssetB,
+        AztecTypes.AztecAsset calldata _outputAssetA,
+        AztecTypes.AztecAsset calldata _outputAssetB,
+        uint256 _totalInputValue,
+        uint256,
+        uint64 _auxData,
+        address _rollupBeneficiary
+    )
+        external
+        payable
+        override(BridgeBase)
+        onlyRollup
+        returns (
+            uint256 outputValueA,
+            uint256,
+            bool
+        )
+    {
+        if (_totalInputValue == 0) {
+            revert ErrorLib.InvalidInputAmount();
+        }
+
+        _loadPools();
+
+        PoolInfo memory selectedPool;
+
+        if (
+            _inputAssetA.assetType == AztecTypes.AztecAssetType.ERC20 &&
+            _outputAssetA.assetType == AztecTypes.AztecAssetType.VIRTUAL
+        ) {
+            // deposit
+            if (!pools[_inputAssetA.erc20Address].exists) {
+                revert ErrorLib.InvalidInputA();
+            }
+
+            selectedPool = pools[_inputAssetA.erc20Address];
```

Comment:
Move this line above line 137 and on line 137 use selectedPool instead. That will save multiple SLOADS and save gas since the struct info is in multiple storage slots.

Reply:
Resolved